x/gov/proposals.go

```diff
@@ -9,142 +9,98 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// Proposal interface
-type Proposal interface {
-	GetProposalID() uint64
-	SetProposalID(uint64)
+// Proposal is a struct used by gov module internally
+// embedds ProposalContent with additional fields to record the status of the proposal process
+type Proposal struct {
```

Comment:
Unclear to me how this is used by the other proposal types. Can we have some sort of documentation about how `Proposal`, `ProposalContent` and `{{ Type }}Proposal` are related?

Reply:
Also, I wonder if this should be called `ProposalMetadata`

Reply:
++ on more documentation. I _do_ think that this struct makes sense to be called `Proposal`

Reply:
OK, I don't have a strong opinion